sycl/include/CL/sycl/handler.hpp

```diff
@@ -166,6 +146,66 @@ checkValueRangeImpl(ValT V) {
 }
 #endif
 
+template <typename KernelType, typename LambdaArgType>
+std::enable_if_t<std::is_same<LambdaArgType, void>::value, bool>
+checkKernelArgTypesIgnoreFirstVoid() {
+  return detail::check_kernel_arg_types<KernelType, kernel_handler>();
+}
+
+template <typename KernelType, typename LambdaArgType>
+std::enable_if_t<!std::is_same<LambdaArgType, void>::value, bool>
+checkKernelArgTypesIgnoreFirstVoid() {
+  return detail::check_kernel_arg_types<KernelType, LambdaArgType,
+                                        kernel_handler>();
+}
+
+template <typename TransformedArgType, typename KernelType, int Dims,
+          typename Enable = void>
+class RangeRoundedLambda;
+
+template <typename TransformedArgType, typename KernelType, int Dims>
+class RangeRoundedLambda<
+    TransformedArgType, KernelType, Dims,
+    typename std::enable_if_t<detail::check_kernel_arg_types<
+        KernelType, TransformedArgType, kernel_handler>()>> {
+public:
+  RangeRoundedLambda(KernelType KernelFunc, range<Dims> NumWorkItems)
+      : MKernelFunc(KernelFunc), MNumWorkItems(NumWorkItems) {}
+
+  void operator()(TransformedArgType Arg) const {
```

Comment:
This function should take kernel_handler and pass it to user lambda.

Reply:
Fixed